viewer/script.js

```diff
@@ -0,0 +1,145 @@
+class ImageRotator {
+    _urls = [];
+    _urlIndex = 0;
+
+    PreviousImage() {
+        if (this._urlIndex > 0) {
+            this._urlIndex--;
+        } else {
+            this._urlIndex = this._urls.length - 1;
+        }
+        this.showImage();
+    }
+
+    NextImage() {
+        if (this._urlIndex < this._urls.length - 1) {
+            this._urlIndex++;
+        } else {
+            this._urlIndex = 0;
+        }
+        this.showImage();
+    }
+
+    setImages(urls) {
+        this._urls = urls;
+        this._urlIndex = 0;
+        this.showImage();
+    }
+
+    showImage() {
+        let url = this._urls[this._urlIndex];
+        if (!url || url === null || url === undefined) {
+            this.viewer.style.display = 'none';
+        } else {
+            this.viewer.src = url;
+            this.viewer.alt = `URL: ${url}`; // Set alt parameter to show the URL of the picture
+            this.viewer.style.display = 'block';
+        }
+    }
+
+    constructor(element) {
+        this.viewer = element;
+    }
+}
+
+class SwipeHandler {
+    _startX
+    _endX
+    //this sets the minimum swipe distance, to avoid noise and to filter actual swipes from just moving fingers
+    _treshold = 100;
+
+//Function to handle swipes
+    _handleTouch() {
+        //calculate the distance on x-axis and o y-axis. Check wheter had the great moving ratio.
+        var xDist = this._endX - this._startX;
+        if (Math.abs(xDist) > this._treshold) {
+            if (xDist > 0) {
+                this.onSwipeRight();
+            } else {
+                this.onSwipeLeft();
+            }
+        }
+    }
+
+    onSwipeLeft
+    onSwipeRight
+
+    constructor() {
+        let viewer = document.getElementById('viewer');
+        viewer.addEventListener('touchstart', (event) => {
+            this._startX = event.touches[0].clientX;
+
+        })
+        viewer.addEventListener('touchend', (event) => {
+            this._endX = event.changedTouches[0].clientX;
+            this._handleTouch()
+
+        })
+    }
+}
+
+let imageRotator
+let swipeHandler
+window.onload = function () {
+    imageRotator = new ImageRotator(document.getElementById('viewer'));
```

Comment:
Since constructor doesn't do anything (which is good), you can just pass the selector there (`viewer`) and left the `ImageRotator` instance the job for finding it when its needed. That way, you won't need the `window.onload` event handler.